subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/store/SubscriptionsDataStore.kt

```diff
@@ -18,28 +18,76 @@ package com.duckduckgo.subscriptions.impl.store
 
 import android.content.SharedPreferences
 import androidx.core.content.edit
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
 
 interface SubscriptionsDataStore {
+
+    // Auth
     var accessToken: String?
     var authToken: String?
     var email: String?
     var externalId: String?
+
+    // Subscription
     var expiresOrRenewsAt: Long?
+    var startedAt: Long?
     var platform: String?
+    var status: String?
+    var entitlements: String?
+    var productId: String?
+
     fun canUseEncryption(): Boolean
 }
 
-class SubscriptionsEncryptedDataStore(
+@ContributesBinding(AppScope::class)
+@SingleInstanceIn(AppScope::class)
+class SubscriptionsEncryptedDataStore @Inject constructor(
     private val sharedPrefsProv: SharedPrefsProvider,
 ) : SubscriptionsDataStore {
-
     private val encryptedPreferences: SharedPreferences? by lazy { encryptedPreferences() }
-
-    @Synchronized
     private fun encryptedPreferences(): SharedPreferences? {
         return sharedPrefsProv.getSharedPrefs(FILENAME)
     }
 
+    override var productId: String?
+        get() = encryptedPreferences?.getString(KEY_PRODUCT_ID, null)
+        set(value) {
+            encryptedPreferences?.edit(commit = true) {
+                if (value == null) {
```

Comment:
I'll leave it with `putString` only and delete the remove bit I don't think it matters if the key exists or if they are null so both work. For longs I'm gonna keep it because `putLong` doesn't take a null although I could change to 0L in that case.